nncf/quantization/algorithms/weight_compression/weight_lowering.py

```diff
@@ -358,6 +358,12 @@ def do_int_quantization(
     """
     assert config.is_integer(), "The function supports integer quantization only"
     group_size = config.group_size
+    is_asym = config.mode in [CompressWeightsMode.INT8_ASYM, CompressWeightsMode.INT4_ASYM]
+    if is_asym and (precomputed_scale is None) != (precomputed_zero_point is None):
+        raise ValueError(
```

Comment:
Is it working with data free approach for INT4_ASYM?

Reply:
During data-free asymmetric compression, both `precomputed_scale` and `precomputed_zero_point` are None, so the if condition does not hold and the exception is not raised.